### ISSUE TITLE
Rename struct fields with a prefix/suffix

### ISSUE THREAD

**Issue by indigoviolet** [pola-rs/polars#5169]
### Problem description

This would be nice to do without having to manually pull out the fields; and allows unnesting the struct with some indication of the "namespace".

**Comment by ritchie46** [comment]
I agree. I would accept a PR that enables this under `struct.suffix/ struct.prefix`.

**Comment by stinodego** [comment]
As mentioned in https://github.com/pola-rs/polars/pull/12281#pullrequestreview-1717106373:

As we have recently centered all renaming functionality under the `name` namespace, I think renaming struct fields should also fall under this namespace.

Perhaps `.name.struct_prefix` / `.name_struct_suffix` ? Or Perhaps `.name.prefix_fields` / `.name.suffix_fields` ?

**Comment by ion-elgreco** [comment]
@stinodego shouldn't https://pola-rs.github.io/polars/py-polars/html/reference/expressions/api/polars.Expr.struct.rename_fields.html this also be moved out of the struct?

**Comment by cmdlineluser** [comment]
Hmm yeah, that's a little awkward.

The main usage of `.struct.prefix` seems to be as a building block for  "unnest_all" / "pd.json_normalize" type behaviour.

Perhaps if something more along those lines existed, there would be little use for this.

The suffixing could then probably be done at the `.col()` level after unnesting if needed.
